### PR TITLE
Sink will now actually apply water to your hands

### LINKED ISSUE #17246: Greys can wash their hands in a sink
but god forbid they wash their hands in a shower.
massive bug I want my hands to be BURNED OFF

### PR DESCRIPTION
**Pull request by ghost**
Note: All line changes past from mop_act() and on are formatting-only (line 505/518)

Yep, it's happening. Greys beware, there's WATER in the sinks. Don't put it on your hands or your face might melt

* Sinks now cause the water TOUCH effect to people using them to wash their hands (yes, even through gloves). This is bad news to a certain species which doesn't like having water on their skin
* Being incapacitated (probably due to the previous point) while using a sink will now interrupt the glorious hand washing

Fixes #17246

:cl:
 * rscadd: Sinks now cause the water TOUCH effect to people using them to wash their hands (yes, even through gloves). This is bad news to a certain species which doesn't like having water on their skin
 * bugfix: Being incapacitated (probably due to the previous point) while using a sink will now interrupt the glorious hand washing

### CODE REVIEW

**Comment by Irradiation** [comment]
Spray the Gray, race war now!

**Comment by brndd** [comment]
>even through gloves

downtooted. my immersion.

**Comment by ghost** [comment]
I mean, the sink also washes your hands through your gloves, even hardsuit gauntlets, so one can assume they're either removed during washing, or it's magic water that goes through armor and souls

**Comment by DrSnips** [comment]
Im pretty sure it just cleaned the gloves before if you had them on and messy hands.

**Comment by brndd** [comment]
Sinks clean your gloves. I don't know if it *also* cleans your hands underneath them, because I've never had gloves on top of dirty hands. Either way water should not go through latex gloves. It's unimmersive.

**Comment by SonixApache** [comment]
>through gloves

🤔 
So I can't wash my gloves while they are on my hands anymore? That's kind of dumb, isn't it?
Does this take permeability into account and all that?

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -487,35 +492,46 @@
...
+	if(M.incapacitated() || !Adjacent(M))
+		return
 
 	M.clean_blood()
 	if(ishuman(M))
-		M:update_inv_gloves()
-	for(var/mob/V in viewers(src, null))
-		V.show_message("<span class='notice'>[M] washes their hands using \the [src].</span>")
+		var/mob/living/carbon/human/H = M
+		if(H.gloves)
```
Comment (ShiftyRail) [inline code/game/objects/structures/watercloset.dm:512]: I would condense this and the following line in a single `if`.

File: code/game/objects/structures/watercloset.dm [hunk 2]
```diff
@@ -453,7 +453,12 @@
 	icon_state = "sink"
 	desc = "A sink used for washing one's hands and face."
 	anchored = 1
-	var/busy = 0 	//Something's being washed at the moment
+	var/busy = 0 //Something's being washed at the moment
+	var/obj/item/weapon/reagent_containers/glass/beaker/water/watersource = null //The liquid dispensed by the sink, normally water, lifted from the shower
```
Comment (ShiftyRail) [inline code/game/objects/structures/watercloset.dm:457]: There _has_ to be a better way to handle this than an invisible infinte (?) beaker
You also need to set this to qdel & set this to null in `Destroy()`

Comment (ghost) [reply code/game/objects/structures/watercloset.dm:457]: That would be the solution the shower took, and yeah

Comment (ShiftyRail) [reply code/game/objects/structures/watercloset.dm:457]: Make a `reagents` datum on the sink and use it directly 
Rather than using `my_beaker.reagents`

**Comment by oversane** [comment]
can i put potassium on my hands too

**Comment by Kurfursten** [comment]
fix it

**Comment by Intigracy** [comment]
fix it

**Comment by brndd** [comment]
>EVEN THROUGH GLOVES

**Comment by Kurfursten** [comment]
You can request this reopened if you want to work on it

**Pull request closed without merging**